### PR TITLE
fix: data types of archived notifications table to match notifications table

### PR DESCRIPTION
**Pull request by kshitij-k-osmosys**
### Pre-requisites

- [x] I have gone through the Contributing guidelines for [Submitting a Pull Request (PR)](https://github.com/OsmosysSoftware/osmo-x/blob/main/CONTRIBUTING.md#-submitting-a-pull-request-pr) and ensured that this is not a duplicate PR.
- [x] I have performed unit testing for the new feature added or updated to ensure the new features added are working as expected.
- [ ] I have added/updated test cases to the [test suite](https://github.com/OsmosysSoftware/osmo-x/blob/main/apps/api/OsmoX.postman_collection.json) as applicable.
- [ ] I have performed preliminary testing using the [test suite](https://github.com/OsmosysSoftware/osmo-x/blob/main/apps/api/OsmoX.postman_collection.json) to ensure that any existing features are not impacted and any new features are working as expected as a whole.
- [ ] I have added/updated the required [api docs](https://github.com/OsmosysSoftware/osmo-x/tree/main/apps/api/docs) as applicable.
- [ ] I have added/updated the `.env.example` file with the required values as applicable.

### PR Details

PR details have been updated as per the given format (see below)

- [x] PR title adheres to the format specified in guidelines (e.g., `feat: add admin login endpoint`)
- [x] Description has been added
- [ ] Related changes have been added (optional)
- [x] Screenshots have been added (optional)
- [ ] Query request and response examples have been added (as applicable, in case added or updated)
- [ ] Documentation changes have been listed (as applicable)
- [ ] Test suite/unit testing output is added (as applicable)
- [ ] Pending actions have been added (optional)
- [ ] Any other additional notes have been added (optional)

### Additional Information

- [x] Appropriate label(s) have been added (`ready for review` should be added if the PR is ready to be reviewed)
- [x] Assignee(s) and reviewer(s) have been added (optional)

_Note: Reviewer should ensure that the checklist and description have been populated and followed correctly, and the PR should be merged only after resolving all conversations and verifying that CI checks pass._

---

**Description:**

Create migration and down file to fix data types of archived notifications table to match notifications table

**Related changes:**
NA

**Screenshots:**
1. Before migration `osmox-local`.notify_archived_notifications
image

2. `osmox-local`.notify_notifications
image

3. After migration `osmox-local`.notify_archived_notifications
image

4. Reverting this migration `osmox-local`.notify_archived_notifications
image

**Query request and response:**
NA

**Documentation changes:**
NA

**Test suite/unit testing output:**
NA

**Pending actions:**
NA

**Additional notes:**
NA

## Summary by CodeRabbit

- **Chores**
	- Updated data types for specific columns in the `notify_archived_notifications` table to enhance data handling.
	- Implemented a rollback strategy for the recent changes to ensure data integrity.

### CODE REVIEW

File: apps/api/src/database/migrations/1732200721350-HotfixArchiveTable.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    // Revert the 'data' column type to its previous type (assuming TEXT)
+    await queryRunner.query(`
+      ALTER TABLE notify_archived_notifications MODIFY COLUMN data TEXT;
+    `);
+
+    // Revert the 'result' column type to its previous type (assuming TEXT)
+    await queryRunner.query(`
+      ALTER TABLE notify_archived_notifications MODIFY COLUMN result TEXT;
+    `);
+  }
```
[the remark this answers is not in the document]
Comment (kshitij-k-osmosys) [reply apps/api/src/database/migrations/1732200721350-HotfixArchiveTable.ts]: fixed 375a9a564766b33935b0107b7ef8224f7e384fb5

File: apps/api/src/database/migrations/1732200721350-HotfixArchiveTable.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class HotfixArchiveTable1732200721350 implements MigrationInterface {
```
Comment (xixas) [inline apps/api/src/database/migrations/1732200721350-HotfixArchiveTable.ts:3]: This is incorrect.
The name of migration should mention what its actually doing(like add table, update table, alter table/column/index etc)...HotFix keyword is used for commits not code.

Comment (xixas) [reply apps/api/src/database/migrations/1732200721350-HotfixArchiveTable.ts:3]: Also once make sure in workbench/dbeaver columns are consistent between the two tables.

**Review by xixas — CHANGES_REQUESTED** [review]

File: apps/api/src/database/migrations/1732200721350-HotfixArchiveTable.ts (hunk 2 above)
Comment (kshitij-k-osmosys) [reply apps/api/src/database/migrations/1732200721350-HotfixArchiveTable.ts]: > The name of migration should mention what its actually doing
Fixed 9647de9d55e8cbffd271ca954ff4ae9e78a11299

> Also once make sure in workbench/dbeaver columns are consistent between the two tables.
Have added the screenshots, kindly check that once please

**Review by xixas — APPROVED** [review]

**Pull request merged**